src/main/java/org/wildfly/security/auth/client/AuthenticationContextConfigurationClient.java

```diff
@@ -134,7 +134,7 @@ public AuthenticationConfiguration getAuthenticationConfiguration(URI uri, Authe
             final IdentityCredentials privateCredentials = securityIdentity.getPrivateCredentials();
             final IdentityCredentials publicCredentials = securityIdentity.getPublicCredentials();
             // private overrides public
-            configuration = configuration.useForwardedIdentity(null).usePrincipal(securityIdentity.getPrincipal()).useCredentials(publicCredentials.with(privateCredentials));
+            configuration = configuration.useForwardedIdentity(securityIdentity.getPrincipal() == AnonymousPrincipal.getInstance() ? forwardSecurityDomain : null).usePrincipal(securityIdentity.getPrincipal()).useCredentials(publicCredentials.with(privateCredentials));
```

Comment:
Yeah, I got that when I first saw. Changing ```RemoteOutboundConnectionService``` seems to be the more correct approach.

One of the things that came in my mind at this regard was to avoid keeping a reference to ```AuthenticationConfiguration``` in ```RemoteOutboundConnectionService``` and just obtain it everytime, directly from the ```AuthenticationContext```.